core/src/Kokkos_Serial.hpp

```diff
@@ -549,10 +549,11 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
     static_assert(Kokkos::is_view<HostViewType>::value,
                   "Kokkos::Serial reduce result must be a View");
 
-    static_assert(
-        Kokkos::Impl::MemorySpaceAccess<typename HostViewType::memory_space,
-                                        Kokkos::HostSpace>::accessible,
-        "Kokkos::Serial reduce result must be a View in HostSpace");
+    static_assert(Kokkos::Impl::MemorySpaceAccess<
+                      typename HostViewType::memory_space,
+                      typename Kokkos::Serial::memory_space>::accessible,
+                  "Kokkos::Serial reduce result must be a View in "
+                  "Serial::memory_space (i.e. HostSpace)");
```

Comment:
I would go for `Kokkos::SpaceAccessibility<Kokkos::Serial, HostViewType::memory_space>::accessible` for readability

Reply:
This change is not correct. Note this tests explicitly whether the reduction result is accessible from the host process (for which HostSpace not Serial::memory_space is the semantic synonym. Say you want to use a copy of Serial later with a custom memory space which fails if you access it outside of the actual kernel dispatch loop, the test needs to be for HostSpace. If you gave that special memory space as the result place the code would crash.